colmena/value_server/server.py

```diff
@@ -0,0 +1,173 @@
+        if len(self.data) >= self.maxsize:
+            lru_key = self.lru.pop()
+            del self.data[lru_key]
+        self.lru.insert(0, key)
+        self.data[key] = value
+
+
+class ValueServer:
+    """Wrapper around a Redis Client for interacting with the value server"""
+    def __init__(self, hostname: str, port: int):
+        """
+        Args:
+            hostname (str): hostname of Redis server
+            port (int): port of Redis server
+        """
+        self.redis_client = redis.StrictRedis(
+            host=hostname, port=port, decode_responses=True)
+        self.cache = LRUCache(LRU_CACHE_SIZE)
+
+    def is_cached(self, key: str, strict: bool = False) -> bool:
+        """Check if key is cached locally
+
+        Args:
+            key (str)
+            strict (bool): if True, cached value must be as new as remote
+
+        Returns:
+            bool
+        """
+        if self.cache.exists(key):
+            if strict:
+                redis_timestamp = float(self.redis_client.get(key + '_timestamp'))
+                cache_timestamp = self.cache.get(key)[0]
+                return cache_timestamp >= redis_timestamp
+            return True
+        return False
+
+    def get(self,
+            key: str,
+            serialization_method: Union[str, SerializationMethod] = SerializationMethod.PICKLE,
+            strict: bool = False
+    ) -> Optional[object]:
+        """Get object by key from value server
+
+        Args:
+            key (str)
+            serialization_method (SerializationMethod): serialization method
+                to use for deserializing object
+            strict (bool): return most recent version of item regardless of
+                timestamp value
+
+        Returns:
+            deserialized object corresponding to key or None if key does not
+            exist
+        """
+        if self.is_cached(key, strict):
+            return self.cache.get(key)[1]
+
+        value = self.redis_client.get(key)
+        if value is not None:
+            timestamp = float(self.redis_client.get(key + '_timestamp'))
+            obj = SerializationMethod.deserialize(serialization_method, value)
+            self.cache.set(key, (timestamp, obj))
+            return obj
+
+        return None
+
+    def set(self,
+            key: str,
+            obj: Any,
+            serialization_method: Union[str, SerializationMethod] = SerializationMethod.PICKLE,
+    ) -> None:
+        """Set object in value server
+
+        Args:
+            key (str)
+            obj (object)
+            serialization_method (SerializationMethod): serialization method
+                to use for serializing object before putting in value server
+        """
+        value = SerializationMethod.serialize(serialization_method, obj)
+        self.redis_client.set(key, value)
+        self.redis_client.set(key + '_timestamp', time.time())
+
+
+def init_value_server(hostname: Optional[str] = None,
```

Comment:
Environment variables makes me a bit nervous. Do you think we should inject the connection information for the server into the result objects?

[Note: Not a blocker for merging]